Sprint-3/todo-list/script.js

```diff
@@ -1,25 +1,72 @@
-function populateTodoList(todos) {
-  let list = document.getElementById("todo-list");
-  // Write your code to create todo list elements with completed and delete buttons here, all todos should display inside the "todo-list" element.
-}
+const removeAllButton = document.getElementById("remove-all-completed");
+const form = document.getElementById("todo-form");
+const input = document.querySelector(".input");
+const todoList = document.getElementById("todo-list");
 
-// These are the same todos that currently display in the HTML
-// You will want to remove the ones in the current HTML after you have created them using JavaScript
 let todos = [
   { task: "Wash the dishes", completed: false },
   { task: "Do the shopping", completed: false },
 ];
 
-populateTodoList(todos);
+// Populate the todo list with initial tasks
+function populateTodoList(todos) {
+  todos.forEach((item) => {
+    addTodoToDOM(item.task, item.completed);
+  });
+}
+
+// Add todo item to the DOM
+function addTodoToDOM(task, completed = false) {
+  const newTodo = document.createElement("li");
+  newTodo.classList.add("todo-item");
+  if (completed) newTodo.classList.add("done");
+  newTodo.innerHTML = `
+    <span class="badge bg-primary rounded-pill">${task}</span>
```

Comment:
This works as long as we trust our users, but there's something called an injection bug, that setting innerHTML using user input can trigger.

We got the value of `task` from the user - they put it in a form. And then we are adding it, exactly as the user typed it, as HTML into our page which will then get rendered by the browser.

Have a read of https://medium.com/@gregadiaz89/dom-innerhtml-vulnerability-8821a03ef2b8 to learn a bit more.

Sometimes we need to use `innerHTML`, but a lot of the time we try to use something like `innerText` to avoid the browser interpreting our text as HTML.

Can you work out how to achieve this same result as you are now, but where we don't include `task` as part of an `innerHTML` property?

Reply:
Thanks, important point, heard about it but never practiced.
I refactored the code.